src/dbg/commands/cmd-gui.cpp

```diff
@@ -10,55 +10,21 @@
 
 bool cbShowThreadId(int argc, char* argv[])
 {
-    duint threadID = 0;
-
-    if(argc > 2)
-    {
-        dprintf(QT_TRANSLATE_NOOP("DBG", "Too many arguments specified. Only 1 argument can be passed, or none.\n"));
-        return false;
-    }
-    else if(argc == 2)
+    if(argc > 1)
     {
-        threadID = strtoll(argv[1], NULL, 10);
-        if(threadID == 0)
-        {
-            dprintf(QT_TRANSLATE_NOOP("DBG", "Could not search for thread ID 0 or argument was not a number in decimal base.\n"));
+        duint threadId = 0;
+        if(!valfromstring(argv[1], &threadId, false))
             return false;
-        }
-    }
-    else if(argc == 1)
-    {
-        GuiShowThreads();
-        return true;
-    }
-
-    THREADLIST threads_head{};
-    DbgGetThreadList(&threads_head);
-    GuiUpdateThreadView(); // To be sure that we have the freshest stuff
 
-    bool threadFound = false;
-    uint32_t row;
-    for(row = 0; row < threads_head.count; row++)
-    {
-        THREADALLINFO currThread = threads_head.list[row];
-        if(currThread.BasicInfo.ThreadId == threadID)
+        SELECTIONDATA newSelection = { threadId, threadId };

```

Comment:
I simplified this by changing from index-based selection to threadId-based selection